src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingSubmissionResource.java

```diff
@@ -95,6 +104,68 @@ public ProgrammingSubmissionResource(ProgrammingSubmissionService programmingSub
         return ResponseEntity.status(HttpStatus.OK).build();
     }
 
+    /**
+     * Trigger the CI build of the given participation.
+     * The build result will be treated as if the user would have done a commit.
+     *
+     * @param participationId of the participation.
+     * @return ok if the participation could be found and has permissions, otherwise forbidden (403) or notFound (404). Will also return notFound if the user's git repository is not available.
+     */
+    @PostMapping(Constants.PROGRAMMING_SUBMISSION_RESOURCE_PATH + "{participationId}/trigger-build")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Void> triggerBuild(@PathVariable Long participationId) {
```

Comment:
There is some duplication here, however I didn't want to mix up the normal build and the instructor build for safety's sake.
Otherwise we would have to identify within the endpoint who the user is and thereby set the corresponding submissionType (Manual vs. Instructor). This would also make it impossible for the instructor to execute a manual build for their own participation (which might sometimes be useful).